src/VestRewardsDistributionJob.sol

```diff
@@ -0,0 +1,168 @@
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Reward Distribution Admin ---
+    function addRewardDistribution(address rewDist, uint256 interval) external auth {
+        if (!distributions.add(rewDist)) revert RewardDistributionExists(rewDist);
+        if (interval < minimumInterval) revert LessThanMinimumInterval(interval);
+        distributionIntervals[rewDist] = interval;
+        emit AddRewardDistribution(rewDist, interval);
+    }
+
+    function removeRewardDistribution(address rewDist) external auth {
+        if (!distributions.remove(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        delete distributionIntervals[rewDist];
+        emit RemoveRewardDistribution(rewDist);
+    }
+
+    function modifyDistributionInterval(address rewDist, uint256 interval) external auth {
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        if (interval < minimumInterval) revert LessThanMinimumInterval(interval);
+        distributionIntervals[rewDist] = interval;
+        emit ModifiedDistributionInterval(rewDist, interval);
+    }
+
+    function work(bytes32 network, bytes calldata args) public {
+        if (!sequencer.isMaster(network)) revert NotMaster(network);
+
+        (address rewDist) = abi.decode(args, (address));
+
+        // prevent keeper from calling random contracts having distribute()
+        if (!distributions.contains(rewDist)) revert RewardDistributionDoesNotExist(rewDist);
+        // ensure that the right time has elapsed
+        if (canDistribute(rewDist)) revert CannotDistributeYet(rewDist);
+        // we omit checking the unpaid amount because if it is 0 it will revert during distribute()
+        uint256 distAmount = VestedRewardsDistributionLike(rewDist).distribute();
+        emit Work(network, rewDist, distAmount);
+    }
+
+    function workable(bytes32 network) external view override returns (bool, bytes memory) {
+        if (!sequencer.isMaster(network)) return (false, bytes("Network is not master"));
+
+        uint256 distributionsLen = distributions.length();
+        if (distributionsLen > 0) {
+            address distributable;
+            for (uint256 i = 0; i < distributionsLen; i++) {
+                address rewDist = distributions.at(i);
+                if (canDistribute(rewDist)) {
+                    uint256 vestId = VestedRewardsDistributionLike(rewDist).vestId();
+                    DssVestWithGemLike dssVest = VestedRewardsDistributionLike(rewDist).dssVest();
+                    uint256 amount = dssVest.unpaid(vestId);
+                    if (amount > 0){
+                        distributable = rewDist;
+                    }
+                }
+            }
+            if (distributable == address(0)) return (false, bytes("No distribution"));
+            return (true, abi.encode(distributable));
+
+        }
+        else {
+            return (false, bytes("No farms"));
+        }
```

Comment:
Nit: this `else` block seems superfluous, you can just have a `return` at the end.
```suggestion
        return (false, bytes("No farms"));
```